src/container/LinkedCellContainer.cpp

```diff
@@ -0,0 +1,390 @@
+            }
+        }
+
+        // update forces within a cell.
+        for(unsigned long j = 0; j < pointCellParticles.size(); ++j){
+            for(unsigned long k = j + 1; k < pointCellParticles.size(); ++k){
+
+                //////////////////////////////////////////////////////////
+                double dist = ArrayUtils::L2Norm((*particles)[pointCellParticles[j]].getX() - (*particles)[pointCellParticles[k]].getX());
+
+                // if the distance is greater than the cutoff, skip the calculation
+                if(dist > cutoff) continue;
+
+                
+                std::array<double, 3> forceIJ = f->force((*particles)[pointCellParticles[j]], (*particles)[pointCellParticles[k]]);
+                
+                std::array<double, 3> v1_force;
+                std::array<double, 3> v2_force;
+                
+                for(int l = 0; l < 3; l++){
+                    v1_force[l] = (*particles)[pointCellParticles[j]].getF()[l] - forceIJ[l];
+                    v2_force[l] = (*particles)[pointCellParticles[k]].getF()[l] + forceIJ[l];
+                }
+
+                (*particles)[pointCellParticles[j]].setF(v1_force);
+                (*particles)[pointCellParticles[k]].setF(v2_force);
+            }
+        }
+
+        
+    }
+
+}
+
+void LinkedCellContainer::updateCellF(const std::vector<int> &v1, const std::vector<int> &v2, bool newton3){
+    for (int i : v1) {
+        for (int j : v2) {
+
+            //////////////////////////////////////////////////////////
+            double dist = ArrayUtils::L2Norm((*particles)[i].getX() - (*particles)[j].getX());
+            if(dist > cutoff) continue; 
+            // if the distance is greater than the cutoff, skip the calculation
+            
+            if (newton3) {
+                std::array<double, 3> forceIJ = f->force((*particles)[i], (*particles)[j]);
+
+                std::array<double, 3> v1_force;
+                std::array<double, 3> v2_force;
+
+                for(int k = 0; k < 3; k++){
+                    v1_force[k] = (*particles)[i].getF()[k] - forceIJ[k];
+                    v2_force[k] = (*particles)[j].getF()[k] + forceIJ[k];
+                }
+
+                (*particles)[i].setF(v1_force);
+                (*particles)[j].setF(v2_force);
+            } else {
+                std::array<double, 3> forceIJ = f->force((*particles)[i], (*particles)[j]);
+                std::array<double, 3> forceJI = f->force((*particles)[j], (*particles)[i]);
+
+                std::array<double, 3> v1_force;
+                std::array<double, 3> v2_force;
+
+                for(int k = 0; k < 3; k++){
+                    v1_force[k] = (*particles)[i].getF()[k] + forceJI[k];
+                    v2_force[k] = (*particles)[j].getF()[k] + forceIJ[k];
+                }
+
+                (*particles)[i].setF(v1_force);
+                (*particles)[j].setF(v2_force);
```

Comment:
There is a difference. In case of newton3==false, the force is computed twice. I actually would like to remove this part as well, but I thought we must keep the possibility to do simulations without Newton 3. Good to know that we don;t, I will completely remove the newton3 flag.